presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2639,22 +2678,22 @@ public TableStatisticsMetadata getStatisticsCollectionMetadataForWrite(Connector
             return TableStatisticsMetadata.empty();
         }
         List<String> partitionedBy = firstNonNull(getPartitionedBy(tableMetadata.getProperties()), ImmutableList.of());
-        return getStatisticsCollectionMetadata(tableMetadata.getColumns(), partitionedBy, false);
+        return getStatisticsCollectionMetadata(tableMetadata.getColumns(), partitionedBy, false, tableMetadata.getTable().getTableName().startsWith(PRESTO_TEMPORARY_TABLE_NAME_PREFIX));
```

Comment:
I don't think this is a reliable way to know if something is a temporary table, since technically any table can start with that prefix

Reply:
You can use Table::PrestoTableType to check temporary table.

Reply:
I see. How about changing it to this:

```
Optional<Table> table = metastore.getTable(tableMetadata.getTable().getSchemaName(), tableMetadata.getTable().getTableName());
return getStatisticsCollectionMetadata(tableMetadata.getColumns(), partitionedBy, false, table.isPresent() && table.get().getTableType() == TEMPORARY_TABLE);
```

Reply:
seems reasonable.